### PR TITLE
Checkout: Make sure Themes, Privacy Protection and Google Apps are behind feature flags

### PR DESCRIPTION
**Pull request by gziolo**
Part of #291.

This PR makes sure it's possible to hide the following features (including buttons & links) behind flags:
* [x] Add Privacy Protection
* [x] Add Google Apps
* [x] Add Theme

It also renames `premium-themes` flag to `upgrades/premium-themes` to keep everything consistent.

### Testing
1. Disable the following flags in the config file:
 * `upgrades/checkout`
 * `upgrades/premium-themes`
2. Execute `make run` and go to http://calypso.localhost:3000/.
3. Go to My Sites.
4. Select a Site.
5. Go to Themes.
6. You should see only free themes.
7. Go to Domain Management section.
8. Go to Email tab.
9. There should be no Add Google Apps button.
10. Go back to Domains list.
11. Select domain with Google App purchase, and go to accounts list.
12. It should not be possible to add another account (no button).
13. Go back to Domains list.
14. Select a domain without Privacy Protection.
14. Then go to Privacy Protection page.
15. It should not be possible to Add Privacy Protection (no button)

Now enable flags listed in (1) and repeat all other steps, you should see all action buttons and links and premium themes :)

### CODE REVIEW

File: config/desktop-mac-app-store.json [hunk 1]
```diff
@@ -17,8 +17,6 @@
 		"code-splitting": false,
 		"network-connection": true,
 		"always_use_logout_url": true,
-		"premium-themes": false,
-		"premium-plans": false,
```
Comment (gziolo) [inline config/desktop-mac-app-store.json]: `premium-plans` flag is never used.

**Comment by janm6k** [comment]
The Google Apps section without button to enable it looks quite confusing:

image

I think that we should either hide it completely, or explain why it cannot be enabled. Premium themes are simply not shown when disabled - so lets use the same approach here?

I am not sure how to handle the privacy protection. In that case, user might want to know about it even if he can't buy it in the app.

Or is this just first step and the disabled state will be added later?

**Comment by gziolo** [comment]
To be honest, this is just the very first step. Our goal is to make it possible to disable all flows that could lead to checkout step. We might improve that in the future, but I'm not sure it that is our goal. We should discuss if we want to provide better experience with those flags disabled. Maybe we could simple present disabled button? Thought that still would be confusing.
@peterbutler & @breezyskies what do you think about it?
Definitely, the main goal of this task is to make sure we can proceed with Mac App Store app verification process. We can always improve UX when we are about to release app in the App Store. I'd like to focus on quick app approval first.

File: client/my-sites/themes/theme-options.js [hunk 2]
```diff
@@ -45,7 +46,7 @@ const buttonOptions = {
 		// We don't know where support docs for a given theme on a self-hosted WP install are,
 		// and free themes don't have support docs.
 		isHidden: ( site, theme ) => ( site && site.jetpack ) || ! Helper.isPremium( theme )
-	},
+	}
```
Comment (janm6k) [inline client/my-sites/themes/theme-options.js]: eslint does not produce any errors with the trailing comma, why remove it? I thought that the trailing comma is removed by babel, and I can't find anything about it in our coding standards, did I miss some discussion about it?

**Comment by janm6k** [comment]
> this is just the very first step

With this is mind, the code (apart trailing comma nitpicking) looks good and works  :+1:

File: client/my-sites/themes/theme-options.js (hunk 2 above)
Comment (gziolo) [inline client/my-sites/themes/theme-options.js]: http://eslint.org/docs/rules/comma-dangle.html - you are right, it's no longer an issue and `comma-dangle` is disabled. There is a big chance that Babel takes care of it. Old habits die hard :D

Comment (gziolo) [inline client/my-sites/themes/theme-options.js]: Done :+1:

**Pull request merged**